x-pack/plugins/security_solution/public/management/components/effected_policy_select/utils.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { PolicyData } from '../../../../common/endpoint/types';
+import { EffectedPolicySelection } from './effected_policy_select';
+
+export const GLOBAL_POLICY_TAG = 'policy:all';
+
+export function getArtifactTagsByEffectedPolicySelection(
+  selection: EffectedPolicySelection
+): string[] {
+  if (selection.isGlobal) {
+    return [GLOBAL_POLICY_TAG];
+  }
+  return selection.selected.map((policy) => {
+    return `policy:${policy.id}`;
+  });
+}
+
+export function getEffectedPolicySelectionByTags(
+  tags: string[],
+  policies: PolicyData[]
+): EffectedPolicySelection {
+  if (tags.length === 0 || tags.find((tag) => tag === GLOBAL_POLICY_TAG)) {
+    return {
+      isGlobal: true,
+      selected: [],
+    };
+  }
+  const selected: PolicyData[] = tags.reduce((acc, tag) => {
+    // edge case: a left over tag with a non-existed policy
+    // will be removed by veryfing the policy exists
+    const id = tag.split(':')[1];
```

Comment:
You are assuming all tags will be a policy tag. As the conversation we had yesterday, we can use this tags for other purposes in the future. I think is safer to check if the first item is a `policy` and then take the `id` as you are doing.

Reply:
I'll leave this helper function as it is right now because it is suppose to be `getArtifactTagsByEffectedPolicySelection`. I'll add the logic to preserve the previous tags in the place that is using this function.

Reply:
@dasansol92 I pushed a commit to preserve previous tags

Reply:
cool, thanks for change that!